coffee_core/src/nurse/chain.rs

```diff
@@ -48,16 +50,18 @@ pub struct RecoveryChainOfResponsibility {
 impl RecoveryChainOfResponsibility {
     pub async fn new() -> Result<Self, CoffeeError> {
         Ok(Self {
-            handlers: Vec::new(),
+            handlers: vec![Arc::new(GitRepositoryLocallyAbsentStrategy)],
         })
     }
 
-    pub async fn scan(&self) -> Result<(), CoffeeError> {
+    pub async fn scan(&self) -> Result<CoffeeNurse, CoffeeError> {
         for handler in self.handlers.iter() {
-            if let Some(strategy) = handler.clone().can_be_apply().await? {
-                strategy.patch().await?;
+            if let Some(strategy) = handler.clone().can_be_applied().await? {
+                return strategy.patch().await;
```

Comment:
```suggestion
                return strategy.patch().await;
```

not sure that this is good, a chan is a sequence of elements and you are returning to the first one that can be applied. 

But maybe the strategy that you have in mind to implement works in this way

Reply:
`can_be_applied` function should ideally serve as an indicator for applying a strategy if and only if the specific case it is designed to handle is encountered. 

I haven't really thought of the idea that we might have 2 or more cases at the same time.
We should be able to fix all the problems with running `coffee nurse` once 

Good catch!

Reply:
>I haven't really thought of the idea that we might have 2 or more cases at the same time.

This is a common case for a chain of responsibility but looks like a good thing to experiment with and then return a vector of operation done by coffee.

So maybe the return should be a push_bach in the vector?